torch/csrc/jit/autodiff.cpp

```diff
@@ -837,13 +867,21 @@ static void lambdaLiftReverse(Gradient& grad_desc, ReverseDetails& rev_info) {
   std::unordered_map<Value*, size_t> capture_to_formal_index;
   const auto& add_capture = [&](Value* captured) {
     capture_to_formal_index[captured] = reverse_block->inputs().size();
-    reverse_block->addInput()->copyMetadata(captured);
+    auto new_input = reverse_block->addInput()->copyMetadata(captured);
+    GRAPH_DEBUG(
+        "Capturing ",
+        captured->debugName(),
+        " as ",
+        new_input->debugName(),
+        " for an embedded backprop block");
   };
   for (auto& offset : grad_desc.df_input_captured_inputs)
     add_capture(graph.inputs()[offset]);
   for (auto& offset : grad_desc.df_input_captured_outputs)
     add_capture(graph.outputs()[offset]);
 
+  GRAPH_DUMP(" forward propagation: ", &graph);
+  GRAPH_DEBUG(" backward propagation: ", *(reverse_block->owningNode()));
```

Comment:
nit: no one calls this forward propagation. Let's do either forward/backward or primal/reverse

Reply:
thanks! will fix!